pandas/tests/internals/test_api.py

```diff
@@ -0,0 +1,11 @@
+"""
+Tests for the pseudo-public API implemented in internals/api.py and exposed
+in core.internals
+"""
+
+from pandas.core import internals
+from pandas.core.internals import api
+
+
```

Comment:
can you also assert the entire namespace of api itself e.g. `[name for name in dir(api) if not name.startswith('_')]` and then compare to a hard coded list.

Reply:
i dont think we want to hard-code the existing list bc whats in there is somewhat haphazard.  Once we hear back from all the relevant parties on #40226 we'll be able to make that list

Reply:
i would fix it now with all of the symbols that are currently exposed

Reply:
picking my battles; updated.